README.md
# site-amanda-boaviagem
[![Netlify Status](https://api.netlify.com/api/v1/badges/198c8a6e-b87c-49d0-809d-ff81959891df/deploy-status)](https://app.netlify.com/sites/epic-pare-6a511b/deploys)

Projeto referente ao site da escritora Amanda Boaviagem. 

## Dependências/stack:

* Node.JS 16+
* React 18
* Next JS 13+

Após o clone do projeto, instalar as dependências:

`$ npm install`

## Executando a aplicação

```shell

# Executar a aplicação local (acessar http://localhost:3000)
$ npm run dev

# Executar apenas o lint
$ npm run lint

# Preparando a aplicação para deploy (Aplicação estará disponível na pasta out/)
$ npm run build

# Executando a aplicação local com compilação otimizada para produção (acessar http://localhost:3000)
$ npm run start

# Executando procedimentos de release, changelog, etc.
$ npm run release

```

## Gerenciamento de conteúdo

Todo o conteúdo do site se apresenta de forma estática no diretório public/data. Todos os arquivos presentes nesta pasta possuem a mesma estrutura:

```javascript

const Data = {
    items: [
        {
            "id": 1,
            "title": "Lorem Ipsum",
            "foo": "bar"
        }
    ]
}

```

Cada arquivo de data pode possuir atributos específicos. Consultar cada arquivo para mais informações sobre a estrutura de dados. 

Tomemos como exemplo o arquivo que armazena os dados da loja (DataStore). Para adicionar novas entradas na loja basta acessar o arquivo public/data/DataStore e incluir um novo objeto conforme o template abaixo. Caso o arquivo possua apenas dois itens, nao sera habilitada a opcao de "ver mais" que leva para a pagina exclusiva da loja.

```javascript

const DataStore = {
    items: [

        ...

        {
            "id": 4,
            "title": "Amor nos Tempos de Quarentena - Amanda Boaviagem",
            "description": "Amor e Morte andam lado a lado. O dia a dia da jovem médica Manuela no hospital em que trabalha em tempos de pandemia do Covid19. Flashes de um passado não tão distante mas feliz, a lua de mel dela e Pedro na Riviera Maya, nas cálidas águas do Caribe. O medo e a insegurança que sente ao ter de encarar a pandemia sozinha. O marido preso em outro país sem poder voltar ao Brasil. Amizades, sonhos, descobertas. Uma luz em sua vida quando tudo parecia perdido. Um antídoto para a pandemia? Acompanhe Manuela nesses tempos difusos (e confusos) em que a verdade é desmentida e o que era solução se torna problema. Sua sanidade mental está sendo posta à prova a todo o momento nessa pandemia? Manuela está pronta pro jogo e disposta a arriscar tudo. E você? O que você seria capaz de fazer por amor?",
            "image": "/images/session03/book1.png",
            "linkAmazon": "https://www.amazon.com.br/gp/product/B095L64BDN/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=B095L64BDN&linkCode=as2&tag=amandaboaviag-20&linkId=6c8777f4c063d10938dc2ffcbe3883d5",
            "linkUiclap": "https://loja.uiclap.com/titulo/ua8461/",
            "direction": "right"
        }
    ]
}

```

De forma análoga, para adicionar novo banner deve ser alterado o arquivo DataIntro adicionando uma entrada no objeto bannersDesktop e outra no objeto bannersMobile com a imagem do tamanho certo para mobile. Para um novo depoimento, deve ser alterado o arquivo DataTestimonials adicionando uma nova entrada no objeto Images e ImagesMobile com o endereço equivalente para a imagem desktop e para a imagem mobile.

**OBS:** Para criar a imagem do tamanho mobile deve-se acessar o canva e cortar a imagem no tamanho para post do instagram. Extensão preferencial de qualquer imagem: *SVG*

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [3.0.0](https://github.com/giovanebribeiro/site-amanda-boaviagem/compare/v2.0.0...v3.0.0) (2023-07-25)


### ⚠ BREAKING CHANGES

* A troca de tecnologia foi completa, de modo que a
estrutura atual é incompatível com estruturas anteriores.

### Features

* reestruturação completa do site ([8c29ce2](https://github.com/giovanebribeiro/site-amanda-boaviagem/commit/8c29ce22489e57754d96598562914337e0dd1c1c))

## [2.0.0](https://github.com/giovanebribeiro/site-amanda-boaviagem/compare/v1.0.0...v2.0.0) (2023-07-25)


public/data/DataStore.js
const DataStore = {

    items: [
        {
            "id": 1,
            "title": "Amor nos Tempos de Quarentena - Amanda Boaviagem",
            "description": "Amor e Morte andam lado a lado. O dia a dia da jovem médica Manuela no hospital em que trabalha em tempos de pandemia do Covid19. Flashes de um passado não tão distante mas feliz, a lua de mel dela e Pedro na Riviera Maya, nas cálidas águas do Caribe. O medo e a insegurança que sente ao ter de encarar a pandemia sozinha. O marido preso em outro país sem poder voltar ao Brasil. Amizades, sonhos, descobertas. Uma luz em sua vida quando tudo parecia perdido. Um antídoto para a pandemia? Acompanhe Manuela nesses tempos difusos (e confusos) em que a verdade é desmentida e o que era solução se torna problema. Sua sanidade mental está sendo posta à prova a todo o momento nessa pandemia? Manuela está pronta pro jogo e disposta a arriscar tudo. E você? O que você seria capaz de fazer por amor?",
            "image": "/images/session03/book1.png",
            "linkAmazon": "https://www.amazon.com.br/gp/product/B095L64BDN/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=B095L64BDN&linkCode=as2&tag=amandaboaviag-20&linkId=6c8777f4c063d10938dc2ffcbe3883d5",
            "linkUiclap": "https://loja.uiclap.com/titulo/ua8461/",
            "direction": "right"
        },
        {
            "id": 2,
            "title": "Contrônicas: Histórias Amorfas de Quarentena - Amanda Boaviagem, Claudia Assis, Livia Messias e Poliana Costa",
            "description": "Como você se sente em relação ao covid-19? Quais as dores, medos e inseguranças que isso gerou na sua vida e no seu dia a dia? Nas “Contrônicas: Histórias Amorfas de Quarentena” o herói, ou melhor os heróis e heroínas, somos todos nós, pessoas comuns, na nossa luta diária pra vencer os desafios que essa pandemia que chegou sem avisar trouxe. Quem diria que a ida ao mercado seria um acontecimento, por exemplo? Mas o brasileiro é sempre criativo e... Quer saber? Que tal descobrir você mesmo nessas páginas? Talvez você descubra a solidariedade que aflora nas pessoas, apesar de tudo, a esperança e uma luz no fim do túnel. Talvez você descubra VOCÊ.",
            "image": "/images/session03/book2.jpg",
            "linkAmazon": "",
            "linkUiclap": "https://loja.uiclap.com/titulo/ua13180/",
            "direction": "left"
        },
        {
            "id": 3,
            "title": "Contrônicas: Histórias Amorfas de Quarentena - Amanda Boaviagem, Claudia Assis, Livia Messias e Poliana Costa",
            "description": "Como você se sente em relação ao covid-19? Quais as dores, medos e inseguranças que isso gerou na sua vida e no seu dia a dia? Nas “Contrônicas: Histórias Amorfas de Quarentena” o herói, ou melhor os heróis e heroínas, somos todos nós, pessoas comuns, na nossa luta diária pra vencer os desafios que essa pandemia que chegou sem avisar trouxe. Quem diria que a ida ao mercado seria um acontecimento, por exemplo? Mas o brasileiro é sempre criativo e... Quer saber? Que tal descobrir você mesmo nessas páginas? Talvez você descubra a solidariedade que aflora nas pessoas, apesar de tudo, a esperança e uma luz no fim do túnel. Talvez você descubra VOCÊ.",
            "image": "/images/session03/book2.jpg",
            "linkAmazon": "",
            "linkUiclap": "https://loja.uiclap.com/titulo/ua13180/",
            "direction": "left"
        },
        {
            "id": 4,
            "title": "Amor nos Tempos de Quarentena - Amanda Boaviagem",
            "description": "Amor e Morte andam lado a lado. O dia a dia da jovem médica Manuela no hospital em que trabalha em tempos de pandemia do Covid19. Flashes de um passado não tão distante mas feliz, a lua de mel dela e Pedro na Riviera Maya, nas cálidas águas do Caribe. O medo e a insegurança que sente ao ter de encarar a pandemia sozinha. O marido preso em outro país sem poder voltar ao Brasil. Amizades, sonhos, descobertas. Uma luz em sua vida quando tudo parecia perdido. Um antídoto para a pandemia? Acompanhe Manuela nesses tempos difusos (e confusos) em que a verdade é desmentida e o que era solução se torna problema. Sua sanidade mental está sendo posta à prova a todo o momento nessa pandemia? Manuela está pronta pro jogo e disposta a arriscar tudo. E você? O que você seria capaz de fazer por amor?",
            "image": "/images/session03/book1.png",
            "linkAmazon": "https://www.amazon.com.br/gp/product/B095L64BDN/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=B095L64BDN&linkCode=as2&tag=amandaboaviag-20&linkId=6c8777f4c063d10938dc2ffcbe3883d5",
            "linkUiclap": "https://loja.uiclap.com/titulo/ua8461/",
            "direction": "right"
        },
        
    ]
}

export default DataStore